sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts

```diff
@@ -132,7 +146,9 @@ export class SchemaRegistryClient implements SchemaRegistry {
    * @param schemaId - Unique schema ID.
    * @returns Schema with given ID.
    */
-  async getSchema(schemaId: string, options?: GetSchemaOptions): Promise<Schema> {
-    return this.client.schema.getById(schemaId, options).then(convertSchemaResponse);
+  async getSchema(schemaId: string, options: GetSchemaOptions = {}): Promise<Schema> {
+    return this._tracing.withSpan("getSchema", options, (updatedOptions) =>
+      this._client.schema.getById(schemaId, updatedOptions).then(convertSchemaResponse)
```

Comment:
Totally not for this PR but I noticed we're mixing async and promise chaining. It's fine to do but I wonder if we should go async/await all the way just for the consistency. Do you think it makes sense to create an issue to do that at some point?

Reply:
To understand the ask here, are you seeking consistency across the whole code base for all libraries? because I do not see any use for await in this module.

Reply:
Well I guess what I'm seeing here is an async declaration on line 149 and a function that returns a promise chain. so it just seemed a little confusing.

I was advocating for:

```ts
  getSchema(schemaId: string, options: GetSchemaOptions = {}): Promise<Schema> {
    return this._tracing.withSpan("getSchema", options, async (updatedOptions) => {
               const response = await this._client.schema.getById(schemaId, updatedOptions)
               return convertSchemaResponse(response)
    }

    );
  }
```

etc... sorry for the bad formatting, and probably a missing } here and there, but I just typed it up as an example... 

If there was no `then` I would say this is fine, but we can remove the `async` modifier (but don't have to), but because we have a promise chain here maybe it makes sense to be consistent - promises all the way or async/await all the way (purely syntactic of course). I don't feel strongly about it though, mostly I just think the `then` here can get lost when scanning this code...

Reply:
oh I see, TIL that `async` unlocks using `await` and not just returning a `Promise`, thanks!